Public/Src/Cache/MemoizationStore/Library/Stores/RocksDbMemoizationStore.cs

```diff
@@ -0,0 +1,194 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+extern alias Async;
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.ContractsLight;
+using System.Threading;
+using System.Threading.Tasks;
+using BuildXL.Cache.ContentStore.Distributed.NuCache;
+using BuildXL.Cache.ContentStore.Interfaces.Extensions;
+using BuildXL.Cache.ContentStore.Interfaces.Logging;
+using BuildXL.Cache.ContentStore.Interfaces.Results;
+using BuildXL.Cache.ContentStore.Interfaces.Sessions;
+using BuildXL.Cache.ContentStore.Interfaces.Time;
+using BuildXL.Cache.ContentStore.Interfaces.Tracing;
+using BuildXL.Cache.ContentStore.Tracing.Internal;
+using BuildXL.Cache.MemoizationStore.Interfaces.Results;
+using BuildXL.Cache.MemoizationStore.Interfaces.Sessions;
+using BuildXL.Cache.MemoizationStore.Interfaces.Stores;
+using BuildXL.Cache.MemoizationStore.Sessions;
+using BuildXL.Cache.MemoizationStore.Tracing;
+
+namespace BuildXL.Cache.MemoizationStore.Stores
+{
+    /// <summary>
+    ///     An IMemoizationStore implementation using RocksDb.
+    /// </summary>
+    public class RocksDbMemoizationStore : IMemoizationStore
+    {
+        private const string Component = nameof(RocksDbMemoizationStore);
+
+        private bool _disposed;
+        private IClock _clock;
+        private RocksDbContentLocationDatabaseConfiguration _config;
+        private RocksDbContentLocationDatabase _cldb;

```

Comment:
I think it is time to rename this type to RocksDbCacheDatabase.

I think we have 3 types in the system: ContentXXX, MemoizationXXX and CacheXXX when the last one combines content and metadata.

Reply:
I will do this in a separate PR. Since there are 34 references just for ContentLocationDatabase, that would add a lot of noise

---
In reply to: [305162151](https://github.com/microsoft/BuildXL/pull/621#discussion_r305162151) [](ancestors = 305162151)